.github/workflows/cicd.yml

```
@@ -193,27 +193,26 @@ jobs:
     strategy:
       matrix:
         os: ["ubuntu-latest"]
-        python-version: ["3.7", "3.9", "3.10", "3.11"]
+        python-version: ["3.9", "3.10", "3.11"]
         twisted-version: ["21.2", "22.1", "23.8"]
         tox-prefix: ["coverage"]
         optional: [false]
         include:
           - os: "ubuntu-latest"
-            python-version: "pypy-3.8"
+            python-version: "pypy-3.9"
             twisted-version: "23.8"
             tox-prefix: "test"
             optional: false
           - os: "ubuntu-latest"
-            python-version: "pypy-3.9"
+            python-version: "3.12.0"
             twisted-version: "23.8"
             tox-prefix: "test"
             optional: false
-          # Test Python 3.12 but allow it to fail
           - os: "ubuntu-latest"
```

I think that a comment describing this job can help with future updates of these values.

```suggestion
          # Test latest supported Python
          # and latest Twisted release.
          - os: "ubuntu-latest"
```